app/controllers/provider/admin/cms/templates_controller.rb

```diff
@@ -27,7 +27,7 @@ def create
   def update
     @page ||= templates.find(params[:id])
 
-    @page.assign_attributes(template_params, without_protection: true)
+    @page.assign_attributes(template_params)
```

Comment:
Are you sure that this can be safely removed?

I don't quite understand what this `without_protection` does, but it was fixing some long-lived issue which was also intermittent, so not 100% reproducible.
The PR: https://github.com/3scale/system/pull/8701
The related issue: https://github.com/3scale/system/issues/7781

Reply:
This PR is related to this one: https://github.com/3scale/porta/pull/3154, I could have merged them in one PR.

Rails implements two ways to protect from mass-assignment, the old one, **protected attributes**, uses white and black lists in the model (`attr_accessible` and `attr_protected`); and the new one to replace the former, **strong parameters**, moves this logic to the controllers. We are using both mixed. What `without_protection` does is bypass the security enforced by `attr_accessible` and `attr_protected`  at the [Layout](https://github.com/3scale/porta/blob/1c8737422ff5848567224b00aa267131e6378bf4/app/models/cms/layout.rb#L3) model, so the user could send any arbitrary attribute in the request, and if it turns out to be a column name, the column would be updated.

That's a security hole, but why would they do that? because the `Layout` model is enforcing a strict mass assignment protection [here](https://github.com/3scale/porta/blob/1c8737422ff5848567224b00aa267131e6378bf4/app/models/cms/layout.rb#L5), so in the case the user sends any attribute which is not white listed, it raises an `ActiveModel::MassAssignmentSecurity::Error` instead of ignoring that attribute and writing a warning in the log which is the default and non-intrusive behavior. In order to avoid the exepction, they bypassed completely the security instead of just stop enforcing the strict security. And why is Layout (only Layout, no other model in all the project) enforcing strict security? I don't know!! that's what I tried to understand and explain in https://github.com/3scale/porta/pull/3154.

Anyway, it's in my [plans](https://issues.redhat.com/browse/THREESCALE-9079) to completely remove all old protected attributes security from CMS models and use only strong parameters in controllers. So that `without_protection` will make no effect after that. In fact, Rails 5 doesn't even support protected attributes, so we are using the the [protected_attributes_continued](https://github.com/westonganger/protected_attributes_continued) gem to get all this working.

Reply:
I forgot the original question:

> Are you sure that this can be safely removed?

Yes, because `without_protection` only makes sense because of [this line](https://github.com/3scale/porta/blob/1c8737422ff5848567224b00aa267131e6378bf4/app/models/cms/layout.rb#L5), and we are removing it at https://github.com/3scale/porta/pull/3154

Reply:
Thanks a lot for the thorough explanation @jlledom ! :clap: 

Yeah, I think it would have probably made sense to merge the two PRs together, but no need to do this now.

As for migrating from protected attributes to strong parameters - that's nice - but I guess that might require quite some refactoring, as the same controllers are used for multiple models (e.g. Templates Controller for Pages, Layouts, Built-in pages etc...). Maybe it will be not that bad in the end.

Reply:
Hold my beer!